### DISCUSSION TITLE
Guide / Instructions for making Gerbera and/or Gerbera-Web UI on Kirkwood Armel on Debian Buster

### DISCUSSION THREAD

**Discussion by BobManMasher** [gerbera/gerbera#1148] [Q&A] [answered]
My Winter lockdown project has been jailbreaking various POGO boxes and using them for NAS and media servers around the house.  After trying a number of media/upnp solutions, I hit on Gerbera and used the standard apt-get version (1.1). Having a few stability problems, I followed the manual instructions to download V1.32 which is much more stable but the web UI code  didn't work correctly.  I then downloaded the latest version of the Gerbera web ui pages and copied the files onto my box. I can now control Gerbera and instruct it to do a manual scan but not an automatic one. The Clients option also has a 404 error so I assume that I have either not installed the server pages correctly or there is a new function in the back end of 1.6 that's needed.  Having looked around, I am still confused about how to either instal the Web UI on its own or compile a new version.  My questions are

1) Can I install the latest Web UI independent of the Gerbera core functions (V1.32 Installed) and how should I do it?

2) Are there any instructions or guides for how to make a Gerbera package for the latest version on Debian Buster and Kirkwood architecture.  There is nothing specific in the manual pages and I did try the basic method without success.

Apologies if I have posted this in the wrong area but it seemed the most appropriate.

Regards - Bob

**Comment by KarlStraussberger** [comment, marked as answer]
Thanks for your question and your efforts with gerbera.

- Taking the newer web pages normally does not work, because the server code has to match. In particular, **clients** was added to 1.6 and **config** to 1.7. It is the fastest way to ruin your database.
- For support with any distro we rely on contribution. So feel free to work out the steps to make us available.

**Reply by BobManMasher** [reply]
Thanks Karl,

- I take on board the issues with running a different Web UI
- I think that creating a build is way beyond my capabilities at the moment.  If I had something for the debian platform I could probably work out the the differences/changes needed.  
 
Does anyone know who made the Debian or kirkwood arch versions?

Regards - Bob

**Reply by BobManMasher** [reply]
So here is my guide to building Gerbera 1.6.4. on Pogo Kirkwood Armel architecture boxes running Debian buster. The process is very similar to the UBUNTU guide but you need a few tweaks.

1) You WILL need a swapfile when using 128Mb devices (and probably 256MB too).  Add a HD or SSD but not a USB stick because of the risk of hardware failure.

2) I opted for libnpupnp and libupnpp6 from https://www.lesbonscomptes.com/upmpdcli/ - Follow the build instructions to create Debian packages which you can then install with dpkg.

2a) If you want regular upnp it is available on deb multimedia stable backports

3) Build the latest Taglib [cmake, make -j4] and use 'Make Install' to  install

4) Use Apt-get to install the rest of the dev packages as per dependencies list. It is best to load fmtlib-dev and libspdlog.dev from the regular Buster Backports (see attached spreadsheet for full listing of versions I used)

5) Clone the Gerbera git and edit the CMakeLists.txt file and comment the original version and add the new.

`# set(GERBERA_VERSION "git")`
`set(GERBERA_VERSION "1.7.0-npupnp")`

NB. I use the version number to highlight my build options while testing

and add these lines to make the debian package to the end of the CMakeLists.txt

`SET(CPACK_GENERATOR "DEB")`
`SET(CPACK_DEBIAN_PACKAGE_MAINTAINER "KK")`
`# include (cmake)`
`include(packaging)`

6) This is my Cmake command:

`cmake -g DEB ../gerbera -DWITH_NPUPNP=1 -DWITH_JS=1 -DWITH_MYSQL=1 -DWITH_CURL=1 -DWITH_TAGLIB=1 -DWITH_MAGIC=1 -DWITH_MATROSKA=1 -DWITH_AVCODEC=1 -DWITH_EXIF=1 -DWITH_EXIV2=1 -DWITH_LASTFM=0 -DWITH_FFMPEGTHUMBNAILER=1 -DWITH_INOTIFY=1`

Resolve any dependency issues now!

7) the 'make -j4' will take at least 6 hours - go for a walk, read a book, grab some sleep .....

8) cpack -G DEB will create a debian package file - All being well - no errors. Use dpkg to install.

9) follow the gerbera manual for installation. Create the gerbera user (give the user a home directory e.g. /home/gerbera). Make the /etc/gerbera folder and get the config.xml. Symbolic link the config file:

`ln -s /etc/gerbera/config.xml /home/gerbera/.config/gerbera`

Symbolic link the web and program directories directory:

`ln -s /usr/share/gerbera /usr/local/share`
`ln -s /var/lib/gerbera /var/local/lib`

10) edit config.xml and change the path to 

`<home>/home/gerbera/.config/gerbera</home>`

11) Start gerbera with the standard launch command. The server should start - watch the messages for errors. Check the web interface functions too. when happy that all is good - control-c to get back to shell

`gerbera -c /etc/gerbera/config.xml` 

12) for SystemD users, copy the gerbera.service script into /usr/systemd/system and edit it to correct the path to the gerbera server the use the systemctl command as per the manual to start and stop the server and debug any problems.

`ExecStart=/usr/bin/gerbera -c /etc/gerbera/config.xml`

13) for init.d users, you need a gerbera script which I took from the earlier version which is in the Debian APT library

14) You need to put your new gerbera package on hold to prevent apt-get upgrade  downgrading back to 1.1

`apt-mark hold gerbera`

That should be everything you need.  I have this Gerbera version running on a PogoPlug V2E02 and a V4 Pro quite happily using vlc and bubbleupnp as clients on to a fire stick and chromecast devices.

A big thank you to the developers here for pointing me in the right direction.
